src/tsickle.ts

```diff
@@ -304,7 +304,15 @@ class ClosureRewriter extends Rewriter {
    *     bind patterns.
    */
   typeToClosure(context: ts.Node, type?: ts.Type, destructuring = false): string {
-    if (this.options.untyped) {
+    if (this.options.untyped && !destructuring) {
+      // Note: even in untyped mode, it's important we provide a type signature
+      // for destructured types, because of the case like:
+      //   function foo({bar}: {bar:number})
```

Comment:
What if we have:

```js
function foo({bar}: {bar: SomeTypeFromTs}) {}
```

Will Closure see `SomeTypeFromTs` and error out because that has never been emitted?